### PR TITLE
Basic UI implementation

### PR DESCRIPTION
**Pull request by treenix**
Currently only works with [https://github.com/veloren/conrod](https://github.com/veloren/conrod).

### CODE REVIEW

**Comment by treenix** [comment]
@xMAC94x chunk_size ok?

File: voxygen/src/game.rs [hunk 1]
```diff
@@ -87,10 +97,11 @@ impl Game {
             window,
             camera: Mutex::new(Camera::new()),
             key_state: Mutex::new(KeyState::new()),
+            ui,
         }
     }
 
-    pub fn handle_window_events(&self) -> bool {
+    pub fn handle_window_events(&mut self) -> bool {
```
Comment (xMAC94x) [inline voxygen/src/game.rs]: this mut isnt needed

File: voxygen/src/game.rs [hunk 2]
```diff
@@ -185,7 +196,7 @@ impl Game {
         }
     }
 
-    pub fn render_frame(&self) {
+    pub fn render_frame(&mut self) {
```
Comment (xMAC94x) [inline voxygen/src/game.rs]: this mut isnt needed

File: voxygen/src/game.rs [hunk 3]
```diff
@@ -241,11 +252,18 @@ impl Game {
...
+        self.ui.set_size(size[0] as u32, size[1] as u32);
+        self.ui.set_ui();
+        renderer.render_ui(&self.ui, &size); */
+
         self.window.swap_buffers();
         renderer.end_frame();
     }
 
-    pub fn run(&self) {
+    pub fn run(&mut self) {
```
Comment (xMAC94x) [inline voxygen/src/game.rs]: this mut isnt needed

File: voxygen/src/game.rs [hunk 4]
```diff
@@ -241,11 +252,18 @@ impl Game {
             renderer.render_model_object(&model);
         }
 
+        /* // Draw ui
+        let size = self.window.get_size();
+
+        self.ui.set_size(size[0] as u32, size[1] as u32);
```
Comment (treenix) [inline voxygen/src/game.rs]: @xMAC94x this needs mutably reference

Comment (xMAC94x) [reply voxygen/src/game.rs]: yeah i noticed this afterward, can you change the Ui in line 46 to a Mutex<Ui> to fit in the other structures

**Comment by treenix** [comment]
@xMAC94x This should do it now

**Comment by xMAC94x** [comment]
+1 for the coding from my side, i will test it locally

**Comment by xMAC94x** [comment]
+1 tested it locally, @zesterer whats your opinion ?

**Comment by treenix** [comment]
@zesterer I refactored the code, please check it

**Comment by zesterer** [comment]
@treenix Looks good. Is `GfxRenderer` a good name for that type? Might be better to just keep it as-is and use `gfx::Renderer`. Aliasing types can cause confusion.

**Pull request merged**